optimade/models/jsonapi.py

```diff
@@ -124,6 +124,9 @@ class Error(BaseModel):
         description="a meta object containing non-standard meta-information about the error.",
     )
 
+    def __hash__(self):
```

Comment:
I don't know how anything worked before. Technically pydantic `BaseModels` are not hashable, so they can't put added to sets. There's several places where `OptimadeError` a subclass of `Error` from `jsonapi.py` is added to a set. I kept getting unhashable errors because this was breaking.

Reply:
Ah! This solves some bugs I have had as well when trying to use sets in different situations. Very good. Indeed, I have instead reverted to adding the `unique_items: true` parameter to the OpenAPI JSON manually, for the fields that should be sets.